imports/plugins/core/accounts/client/components/CreateOrEditGroupDialog.js

```diff
@@ -0,0 +1,222 @@
+    },
+    ...initialValues
+  });
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    submitForm();
+  };
+
+  const rolesForSelect = roles.map((role) => ({ value: role.name, label: role.name }));
+
+  return (
+    <Dialog
+      open={isOpen}
+      onClose={onClose}
+      classes={{ root: classes.cardRoot }}
+      fullWidth
+      maxWidth="sm"
+    >
+      <CardHeader
+        action={
+          <IconButton aria-label="close" onClick={onClose}>
+            <CloseIcon />
+          </IconButton>
+        }
+        title={i18next.t("admin.groupCards.createOrUpdateGroupDialog.title")}
+      />
+      <CardContent>
+        <Grid container spacing={1} className={classes.cardContainer}>
+          <Grid item sm={12}>
+            <TextField
+              error={hasErrors(["name"])}
+              fullWidth
+              helperText={getFirstErrorMessage(["name"])}
+              label={i18next.t("admin.groupCards.createOrUpdateGroupDialog.name")}
+              {...getInputProps("name", muiOptions)}
+            />
+          </Grid>
+          <Grid item sm={12}>
+            <TextField
+              error={hasErrors(["slug"])}
+              fullWidth
+              helperText={getFirstErrorMessage(["slug"])}
+              label={i18next.t("admin.groupCards.createOrUpdateGroupDialog.slug")}
+              {...getInputProps("slug", muiOptions)}
+            />
+          </Grid>
+          <Grid item sm={12}>
+            <TextField
+              error={hasErrors(["description"])}
+              fullWidth
+              helperText={getFirstErrorMessage(["description"])}
+              label={i18next.t("admin.groupCards.createOrUpdateGroupDialog.description")}
+              {...getInputProps("description", muiOptions)}
+            />
+          </Grid>
+          <Grid item sm={12}>
+            <Select
+              fullWidth
+              isMulti
+              options={rolesForSelect}
+              onSelection={setSelectedRoles}
+              placeholder={i18next.t("admin.groupCards.createOrUpdateGroupDialog.selectRoles")}
+              value={selectedRoles}
+            />
+          </Grid>
+        </Grid>
+      </CardContent>
+      <CardActions className={classes.cardActions}>
+        <Box>
+          <Button
+            onClick={onClose}
+          >
+            {i18next.t("app.cancel")}
+          </Button>
+        </Box>
+        <Button
+          color="primary"
+          disabled={isSubmitting || !isDirty}
+          variant="contained"
+          onClick={handleSubmit}
+          type="submit"
+        >
+          {isSubmitting ? i18next.t("app.settings.saveProcessing") : i18next.t("app.saveChanges")}
+        </Button>
+      </CardActions>
+    </Dialog>
+  );
+}
+
+CreateOrEditGroup.propTypes = {
+  group: PropTypes.object,
+  isOpen: PropTypes.bool.isRequired,
+  onClose: PropTypes.func.isRequired,
+  onSuccess: PropTypes.func,
+  shopId: PropTypes.string.isRequired
```

Comment:
This generates a prop type error, due to the fact that when React attempt to render the first time, the shopId is not yet available, making it not required mitigates this.
```suggestion
  shopId: PropTypes.string
```